ui/src/index.js

```diff
@@ -3,17 +3,144 @@ import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
 
 import './index.css';
-import App from './App';
+import App from './components/App';
 import configureStore from './configureStore';
 
+import {
+  TABLE_SORT_DIRECTION,
+  defaultSortingOrder,
+  sortableHeaderCellFormatter,
+  tableCellFormatter
+} from 'patternfly-react';
+import * as sort from 'sortabular';
+
+// Point the transform to your sortingColumns. React state can work for this purpose
+// but you can use a state manager as well.
+const getSortingColumns = () => preloadedState.sortingColumns || {};
+
+const sortableTransform = sort.sort({
+  getSortingColumns,
+  onSort: (selectedColumn) => {
+    this.setState({
+      sortingColumns: sort.byColumn({
+        sortingColumns: this.state.sortingColumns,
+        sortingOrder: defaultSortingOrder,
+        selectedColumn
+      })
+    });
+  },
+  // Use property or index dependening on the sortingColumns structure specified
+  strategy: sort.strategies.byProperty
+});
+
+const sortingFormatter = sort.header({
+  sortableTransform,
+  getSortingColumns,
+  strategy: sort.strategies.byProperty
+});
+
 const preloadedState = {
-  apps: [
-    { id: 1000, app: 'App-A', versions: 3, clients: 1245, startups: 'male', birth_year: '1979', actions: null },
-    { id: 1001, app: 'App-B', versions: 4, clients: 655, startups: 'male', birth_year: '1974', actions: null },
-    { id: 1002, app: 'App-C', versions: 1, clients: 970, startups: 'female', birth_year: '1989', actions: null },
-    { id: 1003, app: 'App-D', versions: 6, clients: 255, startups: 'male', birth_year: '1990', actions: null },
-    { id: 1004, app: 'App-E', versions: 5, clients: 120, startups: 'female', birth_year: '1999', actions: null }
-  ]
+  appGrid: {
```

Comment:
Should we be storing the table setup as state? I would have thought that this is presentation rather than state?